packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+    cancel: take(Actions.CANCEL_VERIFICATION),
+    timeout: delay(VERIFICATION_TIMEOUT),
+  })
+
+  if (result === true) {
+    ValoraAnalytics.track(VerificationEvents.verification_complete)
+    Logger.debug(TAG, 'Verification completed successfully')
+  } else if (result) {
+    ValoraAnalytics.track(VerificationEvents.verification_error, { error: result })
+    Logger.debug(TAG, 'Verification failed')
+  } else if (cancel) {
+    ValoraAnalytics.track(VerificationEvents.verification_cancel)
+    Logger.debug(TAG, 'Verification cancelled')
+  } else if (timeout) {
+    ValoraAnalytics.track(VerificationEvents.verification_timeout)
+    Logger.debug(TAG, 'Verification timed out')
+    yield put(showError(ErrorMessages.VERIFICATION_TIMEOUT))
+    yield put(feelessSetVerificationStatus(VerificationStatus.Failed))
+  }
+  Logger.debug(TAG, 'Done verification')
+
+  yield put(refreshAllBalances())
+}
+
+export function* feelessRestartableVerification(initialWithoutRevealing: boolean) {
+  let isRestarted = false
+  while (true) {
+    const withoutRevealing = !isRestarted && initialWithoutRevealing
+    yield call(navigate, Screens.VerificationLoadingScreen, {
+      withoutRevealing,
+    })
+    yield put(feelessResetVerification())
+    yield call(getConnectedAccount)
+    if (isRestarted || (yield select(isFeelessVerificationStateExpiredSelector))) {
+      yield call(feelessFetchVerificationState)
+    }
+
+    const { verification, restart } = yield race({
+      verification: call(feelessDoVerificationFlow, withoutRevealing),
+      restart: take(Actions.FEELESS_RESEND_ATTESTATIONS),
+    })
+
+    if (restart) {
+      isRestarted = true
+      const { status }: FeelessVerificationState = yield select(feelessVerificationStateSelector)
+      ValoraAnalytics.track(VerificationEvents.verification_resend_messages, {
+        count: status.numAttestationsRemaining,
+      })
+    } else {
+      return verification
+    }
+    Logger.debug(TAG, 'Verification has been restarted')
+  }
+}
+
+export function* feelessDoVerificationFlow(withoutRevealing: boolean = false) {
+  Logger.debug(TAG, '@feelessDoVerificationFlow', 'Starting feeless verification')
+  let receiveMessageTask: Task | undefined
+  let autoRetrievalTask: Task | undefined
+
+  try {
+    const [walletAddress, contractKit, e164Number]: [string, ContractKit, string] = yield all([
+      call(getConnectedUnlockedAccount),
+      call(getContractKit),
+      select(e164NumberSelector),
+      put(feelessSetVerificationStatus(VerificationStatus.Prepping)),
+    ])
+
+    let feelessVerificationState: FeelessVerificationState = yield select(
+      feelessVerificationStateSelector
+    )
+
+    const komenciKit = new KomenciKit(contractKit, walletAddress, {
+      url: KOMENCI_URL,
+      token: feelessVerificationState.komenci.sessionToken,
+    })
+
+    // Start by checking again to make sure Komenci is ready. Throws error if not
```

Comment:
Yeah, I was thinking about the instance where there was a long wait between the call that happens in `feelessFetchVerificationState` and the actual verification flow, which would be the user leaving the "start verification" open for a long time without actually pressing "start".

Thinking more about it, it's probably not very practical because the next call would just fail anyways if Komenci is down so there is no UX advantage here (as opposed to the previous one where we could divert users to non-feeless verification). I'll remove this.